bee-protocol/src/worker/solidifier/transaction.rs

```diff
@@ -35,15 +35,18 @@ impl TransactionSolidifierWorker {
 
     // TODO is the index even needed ? We request one milestone at a time ? No PriorityQueue ?
 
-    async fn solidify(&self, hash: Hash, index: u32) -> bool {
+    async fn solidify(&self, hash: Hash, index: MilestoneIndex) -> bool {
         let mut missing_hashes = HashSet::new();
 
-        tangle().walk_approvees_depth_first(
+        traversal::visit_parents_depth_first(
+            &tangle().inner,
             hash,
-            |_| {},
-            |vertex| !vertex.is_solid(),
-            |missing_hash| {
-                missing_hashes.insert(*missing_hash);
+            |_, metadata| metadata.is_solid(),
```

Comment:
Did the `!` disappear ?

Reply:
good catch!

Reply:
how do you like that vertex is gone? :trollface:

Reply:
A dream come true